### PR TITLE
fix for the 1.0.8 upgrade fiasco

### PR DESCRIPTION
**Pull request by maxtaco**
- register as logger, since otherwise service will kill the connection on a "protocol not found" error
- comments and cleanups

### CODE REVIEW

File: go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go [hunk 1]
```diff
@@ -46,7 +46,11 @@ func (d *dispatch) Call(ctx context.Context, name string, arg interface{}, res i
 	d.calls.AddCall(c)
 	defer d.calls.RemoveCall(c.seqid)
 	rpcTags, _ := RpcTagsFromContext(ctx)
-	errCh := d.writer.EncodeAndWrite(ctx, []interface{}{MethodCall, c.seqid, c.method, c.arg, rpcTags})
+	v := []interface{}{MethodCall, c.seqid, c.method, c.arg}
```
Comment (patrickxb) [inline go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go]: why is `ok` from line 48 ignored?

Comment (maxtaco) [inline go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go]: i actually don't need this change, we can back it out

Comment (maxtaco) [inline go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go]: but then i'll have to retest...

actually best to leave it in...

File: go/client/versionfix.go [hunk 2]
```diff
@@ -8,24 +8,73 @@ import (
...
+	// (like 1.0.8)
+	socket, err = g.SocketInfo.DialSocket()
 	if err != nil {
 		return err
 	}
+	xp := libkb.NewTransportFromSocket(socket)
+	srv := rpc.NewServer(xp, libkb.WrapError)
+	gcli := rpc.NewClient(xp, libkb.ErrorUnwrapper{})
+	cli = keybase1.ConfigClient{Cli: gcli}
+	srv.Register(NewLogUIProtocol())
```
Comment (patrickxb) [inline go/client/versionfix.go]: Should LogUIProtocol be contextified?  I know outside of this PR, but...

File: go/client/versionfix.go [hunk 3]
```diff
@@ -8,24 +8,73 @@ import (
...
 func FixVersionClash(g *libkb.GlobalContext, cl libkb.CommandLine) (err error) {
 	var cli keybase1.ConfigClient
 	var ctlCli keybase1.CtlClient
 	var serviceConfig keybase1.Config
+	var socket net.Conn
 
 	g.Log.Debug("+ FixVersionClash")
 	defer func() {
+		if socket != nil {
+			socket.Close()
```
Comment (patrickxb) [inline go/client/versionfix.go]: and `socket = nil`?

File: go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go (hunk 1 above)
Comment (patrickxb) [inline go/vendor/github.com/keybase/go-framed-msgpack-rpc/dispatch.go]: ok.

File: go/client/versionfix.go (hunk 3 above)
Comment (maxtaco) [inline go/client/versionfix.go]: ?  come again ?

File: go/client/versionfix.go (hunk 2 above)
Comment (maxtaco) [inline go/client/versionfix.go]: yes

**Comment by patrickxb** [comment]
LGTM.

Don't worry about LogUIProtocol

We can do that later.

File: go/client/versionfix.go (hunk 2 above)
Comment (patrickxb) [inline go/client/versionfix.go]: Nevermind.  Out of scope.

File: go/client/versionfix.go (hunk 3 above)
Comment (patrickxb) [inline go/client/versionfix.go]: the other place `socket.Close()` happens, it is set to nil right afterwards.  wasn't sure if that should happen after this `socket.Close()` too or not.

**Pull request merged**